composites/Plugin/SnippetEditor/components/SnippetEditor.js

```diff
@@ -352,17 +352,19 @@ class SnippetEditor extends React.Component {
 		// Strip multiple spaces and spaces at the beginning and end.
 		description = stripSpaces( description );
 
-		const mappedData = {
+		let mappedData = {
 			title: this.processReplacementVariables( originalData.title ),
-			url: baseUrl.replace( "https://", "" ) + originalData.slug,
+			url: originalData.slug,
 			description: description,
 		};
 
 		// The mapping by the passed mapping function should happen before measuring.
 		if ( mapDataToPreview ) {
-			return mapDataToPreview( mappedData, originalData );
+			mappedData =  mapDataToPreview( mappedData, originalData );
```

Comment:
I'd like to doublecheck this decision with @atimmer. Is it okay to have only the slug without the base URL at the point where the passed `mapDataToPreview` function is going to do its work? Can we assume the passed `mapDataToPreview` function will never need to change the base URL? And if so, calling it 'url' at this point might make things unclear.

Reply:
Please pass the `baseUrl` in a `context` 3th argument. Its a clean solution and makes it future compatible.